src/test/java/org/mockitousage/matchers/CapturingArgumentsTest.java

```diff
@@ -315,4 +315,31 @@ public void captures_correctly_when_captor_used_on_pure_vararg_method() throws E
         verify(mock).varargs(eq(42), argumentCaptor.capture());
         Assertions.assertThat(argumentCaptor.getValue()).contains("capturedValue");
     }
+
+    @Test
+    public void should_capture_null_single_argument_in_vararg_method() {
+        // given
+        ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
+
+        // when
+        mock.varargs(null);
+
+        // then
+        verify(mock).varargs(argumentCaptor.capture());
+        Assertions.assertThat(argumentCaptor.getValue()).isNull();
+    }
+
+    @Test
+    public void should_capture_null_array_in_method_vararg() {
+        // given
+        ArgumentCaptor<String[]> argumentCaptor = ArgumentCaptor.forClass(String[].class);
+
+        // when
+        String[] nullArray = null;
+        mock.varargs(nullArray);
+
+        // then
+        verify(mock).varargs(argumentCaptor.capture());
+        Assertions.assertThat(argumentCaptor.getValue()).isNull();
```

Comment:
@lukasz-szewc 
I think we speak about two different things here. I mean we should remove the capturing part in case the vararg-array is null not the vararg element that is null. It should work if you simply remove the [*else* part in line 149](https://github.com/mockito/mockito/pull/606/files#diff-6bedbf35a65e3a44c8e6c7f033172013R149). This change should not cause an NPE.

----

@TimvdLippe 

> My expectation as a user would be that the captor does capture null, just like any other null invocation.

My expectation is that a ArgumentCaptor<String> only capture Strings, but if Ian String[] is passed even if it is null then it should capture nothing, this relates to #565.